TYP: np.lib.stride_tricks.sliding_window_view parameter axis typing is inconsistent with documentation and implementation.

### Describe the issue:

In the stub file `numpy/lib/_stride_tricks_impl.pyi`, the definition of `sliding_window_view` states that the `axis` parameter is an optional `SupportsIndex` (so a single axis).
In the documentation, and in the implementation, it accepts a `numpy._typing._ShapeLike`.

Type stubs:
https://github.com/numpy/numpy/blob/bf18717b688fa71203ae73646d877d9fac5ff50a/numpy/lib/_stride_tricks_impl.pyi#L39-L56

Documentation:
https://github.com/numpy/numpy/blob/bf18717b688fa71203ae73646d877d9fac5ff50a/numpy/lib/_stride_tricks_impl.py#L191-L214

Implementation:
https://github.com/numpy/numpy/blob/bf18717b688fa71203ae73646d877d9fac5ff50a/numpy/lib/_stride_tricks_impl.py#L419-L431

Reproducibility:
I created a fresh environment:
```console
mkdir numpy_typing
cd numpy_typing
uv init
uv add numpy
uv add mypy
```

Note: I'm very willing to send a PR to fix this but since it´d be my first contribution to numpy, I wouldn't mind being a bit guided on the process.

### Reproduce the code example:

```python
import numpy as np

x = np.zeros((10, 20, 20))
window_shape = (5,5)
axis = (1,2)
np.lib.stride_tricks.sliding_window_view(x=x, window_shape=window_shape, axis=axis)
```

### Error message:

```shell
main.py:6: error: No overload variant of "sliding_window_view" matches argument types "ndarray[tuple[int,
 int, int], dtype[float64]]", "tuple[int, int]", "tuple[int, int]"  [call-overload]
main.py:6: note: Possible overload variants:
main.py:6: note:     def [_ScalarT: generic[Any]] sliding_window_view(x: _SupportsArray[dtype[_ScalarT]] 
| _NestedSequence[_SupportsArray[dtype[_ScalarT]]], window_shape: int | Iterable[int], axis: SupportsInde
x | None = ..., *, subok: bool = ..., writeable: bool = ...) -> ndarray[tuple[Any, ...], dtype[_ScalarT]]
main.py:6: note:     def sliding_window_view(x: Buffer | _SupportsArray[dtype[Any]] | _NestedSequence[_Su
pportsArray[dtype[Any]]] | complex | bytes | str | _NestedSequence[complex | bytes | str], window_shape: 
int | Iterable[int], axis: SupportsIndex | None = ..., *, subok: bool = ..., writeable: bool = ...) -> nd
array[tuple[Any, ...], dtype[Any]]
Found 1 error in 1 file (checked 1 source file)
```

### Python and NumPy Versions:

Numpy version: 2.4.4
Python version: 3.14.3 (main, Feb 13 2026, 15:31:44) [GCC 15.2.1 20260209]

### Type-checker version and settings:

Type checker used: mypy 1.20.1 (compiled: yes)
Mypy used with default configurarion on new installation.

### Additional typing packages.

_No response_

--- a/numpy/lib/_stride_tricks_impl.pyi
+++ b/numpy/lib/_stride_tricks_impl.pyi
@@ -38,7 +38,7 @@ def as_strided(
 def sliding_window_view(
     x: _ArrayLike[_ScalarT],
     window_shape: int | Iterable[int],
-    axis: SupportsIndex | None = None,
+    axis: int | tuple[int, ...] | None = None,
     *,
     subok: bool = False,
     writeable: bool = False,
@@ -47,7 +47,7 @@ def sliding_window_view(
 def sliding_window_view(
     x: ArrayLike,
     window_shape: int | Iterable[int],
-    axis: SupportsIndex | None = None,
+    axis: int | tuple[int, ...] | None = None,
     *,
     subok: bool = False,
     writeable: bool = False,

